pages/agent/v3/signed_pipelines.md

```diff
@@ -0,0 +1,162 @@
+
+```sh
+buildkite-agent tool keygen --help
+```
+
+### 2. Configure the agents
+
+Next, you need to configure your agents to use the keys you generated. On agents that upload pipelines, add the following to the agent's config file:
+
+```ini
+job-signing-jwks-path=<path to private keyset>
+job-signing-key-id=<the key id you generated earlier>
+job-verification-jwks-path=<path to public keyset>
+```
+
+This ensures that whenever those agents upload steps to Buildkite, they'll generate signatures using the private key you generated earlier. It also ensures that those agents verify the signatures of any steps they run, using the public key.
+
+On instances that verify jobs, add:
+
+```ini
+job-verification-jwks-path=<path to verification keys>
+```
+
+### 3. Sign all steps
+
+So far, you've configured agents to sign and verify any steps they upload and run. But there's another source of steps: the Buildkite dashboard. Rather than defining steps in your repository, you can also define steps in a pipeline's settings. The most common use for this is to have a single step that uploads a pipeline definition from [a YAML file in the repository](/docs/pipelines/defining-steps#step-defaults-pipeline-dot-yml-file). These steps should also be signed.
+
+> 🚧 Non-YAML steps
+> You must use YAML to sign steps configured in the Buildkite dashboard. If you don't use YAML, you'll need to [migrate to YAML steps](/docs/tutorials/pipeline_upgrade) before continuing.
+
+To sign steps configured in the Buildkite dashboard, you need to add static signatures to the YAML:
+
+1. Copy the existing YAML steps definition from the editor in the Buildkite dashboard.
+1. Paste the step definitions into a file on your machine.
+1. Generate a new YAML definition with signatures by running:
+
+    ```sh
+    buildkite-agent pipeline upload --jwks-file-path <path-to-private-keys> --signing-key-id <key-id> --dry-run --format yaml <path-to-pipeline-file>
+    ```
+
+    Replacing the following:
+    * `<path-to-private-keys>` with the path to your private keys.
+    * `<key-id>` with the key ID.
+    * `<path-to-pipeline-file>` with the path to the step definitions you saved from the Buildkite dashboard.
+
+    This outputs the pipeline YAML with the signatures added to the steps.
+
+1. Copy the updated step definitions back into the editor in the Buildkite dashboard.
+1. Save the new definition.
+
+For example, if you had a file called `pipeline.yaml` in the current directory, and you wanted to sign it with the key generated earlier, you'd run:
+
+```bash
+buildkite-agent pipeline upload --jwks-file-path /path/to/private/key.json --signing-key-id my-key-id --dry-run --format yaml pipeline.yaml
+```
+
+## Rotating signing keys
+
+Because signed pipelines use JWKS as their key format, rotating keys is easy.
+
+To rotate your keys:
+
+1. Generate a new key pair.
+1. Add the new keys to your existing key sets. Be careful not to mix public and private keys.
```

Comment:
```suggestion
2. Add the new keys to your existing key sets. Be careful not to mix public and private keys.
```

Reply:
`1.` is correct for both list items. It displays as the right number when rendered and helps make future refactoring simpler. 😊

![CleanShot 2023-11-21 at 09 59 16](https://github.com/buildkite/docs/assets/119824349/44d7d3a9-cc9a-4010-9a5a-9d4b0a46e179)

Reply:
awesome thank you for the clarity :) I'll admit that I did not have it rendered when making this suggestion!